radixdlt-core/radixdlt/src/main/java/com/radixdlt/consensus/ConsensusEvent.java

```diff
@@ -97,4 +97,11 @@
 	 * @return {@link HighQC} associated with event
 	 */
 	HighQC highQC();
+
+	/**
+	 * Dispatch processing to corresponding method in the processor.
+	 *
+	 * @param processor	the processor to use
+	 */
+	void processBy(BFTEventProcessor processor);
```

Comment:
`ConsensusEvent` should not know about who's processing it or how's it being processed. This breaks abstractions. Please remove.

Reply:
But it still doesn't know, BFTEventProcessor is also an abstraction. This is some kind of simpler Visitor pattern to avoid overuse of `if/else` and `instanceOf`. With a sealed interface and pattern matching it becomes less critical though.